uint32 pixel_dist_x